daffodil-core/src/main/scala/org/apache/daffodil/dsom/RuntimePropertyMixins.scala

```diff
@@ -367,9 +367,14 @@ trait ElementRuntimeValuedPropertiesMixin
         case (Implicit, Binary, HexBinary) => (LengthUnits.Bytes, maxLengthLong) // fixed length
         case (Implicit, Text, String) => (lengthUnits, maxLengthLong) // fixed length
         case (Implicit, Text, AnySimpleType) => (lengthUnits, textOutputMinLength) // fixed length
-        case (Explicit, Text, String) => (lengthUnits, minLengthLong)
-        case (Explicit, Binary, HexBinary) => (LengthUnits.Bytes, minLengthLong)
-        case (Explicit, Text, AnySimpleType) => (lengthUnits, textOutputMinLength)
+        case (Explicit, Text, String) => (lengthUnits, 0L)
+        case (Explicit, Binary, HexBinary) => (LengthUnits.Bytes, 0L)
+        case (Explicit, Text, AnySimpleType) => {
+          if (optLengthConstant.isDefined && (optLengthConstant.get < minLengthLong))
+            SDE("Explicit dfdl:length of %d is out of range for facet textOutputMinLength='%d'",
+              optLengthConstant.get, textOutputMinLength)
+          (lengthUnits, 0L)
+        }
```

Comment:
Should this check also be done at runtime if dfdl:lengthKind="explicit" but dfdl:length is an expression? Or is that a validation error that might already handle?

Reply:
Actually, I just double checked and it seems that there already is runtime validation and this check is unnecessary and can be set to (lengthUnits, 0L) like the other cases for lengthKind Explicit